src/libsyntax/parse/parser.rs

```diff
@@ -1198,24 +1198,29 @@ impl<'a> Parser<'a> {
                 };
 
                 let body = match p.token {
-                  token::Semi => {
-                    try!(p.bump());
-                    debug!("parse_trait_methods(): parsing required method");
-                    None
-                  }
-                  token::OpenDelim(token::Brace) => {
-                    debug!("parse_trait_methods(): parsing provided method");
-                    let (inner_attrs, body) =
-                        try!(p.parse_inner_attrs_and_block());
-                    attrs.extend(inner_attrs.iter().cloned());
-                    Some(body)
-                  }
+                    token::Semi => {
+                        try!(p.bump());
+                        if let token::DocComment(_) = p.token {
```

Comment:
You likely want some sort of loop, because lexer may emit multiple `DocComment` in a row.

That being said… I do not think this is the correct way or place to do this kind of thing. Lets see what other people have to say about this.